wall_e/src/WalleModels/models.py

```diff
@@ -1,19 +1,80 @@
+# for type references to own class
+# https://stackoverflow.com/a/33533514
+from __future__ import annotations
+from typing import List
 import datetime
 import logging
 import random
 import time
-
 import pytz
 from asgiref.sync import sync_to_async
 from django.conf import settings
 from django.db import models
 from django.forms import model_to_dict
-
+from .customFields import GeneratedIdentityField
 import django_db_orm_settings
 
 logger = logging.getLogger('wall_e')
 
 
+class BanRecords(models.Model):
+    ban_id = GeneratedIdentityField(primary_key=True, always=True)
+    username = models.CharField(max_length=32, null=False)
+    user_id = models.BigIntegerField(null=False)
+    mod = models.CharField(max_length=32, null=True)
+    mod_id = models.BigIntegerField(null=True)
+    ban_date = models.BigIntegerField(null=True)
+    reason = models.TextField(null=False)
```

Comment:
ill change it, originally i just didn't wanna put a cap on the reason in case someone decided to be detailed. however a normal ban only has 512 char limit so ill just copy that.